### ISSUE TITLE
Convert HR to uids_base

### ISSUE THREAD

**Issue by richardbporter** [uiowa/uiowa#2629]
Originally 12-25 hours estimate for developer work. Does not include possible student work in replacing fontawesome icons with brand-approved versions and the scope has increased with recent work related to events, uiowa_bar, etc.

- [x] #3707
- [x] #3708
- [x] #2611 
- [x] https://github.com/uiowa/uiowa/issues/1606
- [x] Remove https://github.com/uiowa/uiowa/pull/2711/commits/9fe9bd5ea9b5b959116463c2136724755e015572 per https://github.com/uiowa/uiowa/pull/2711
- [x] Remove `config_ignore.settings` from `hr` site split.

## Related clean-up tasks
- [x] #1907 
- [x] #1713 
- [x] Remove https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/sitenow_people/sitenow_people.module#L269-L287 per https://github.com/uiowa/uiowa/pull/2906
- [x] #3719

**Comment by joewhitsitt** [comment]
>Sean Hesler is assuming UHR’s web leadership from a functional standpoint and will continue to utilize Lin and ITS for our technical expertise - Megan Hammes

**Comment by richardbporter** [comment]
The checklist above is not an exhaustive list, more just noting things than need/can be done in conjunction with this.

**Comment by joewhitsitt** [comment]
Overall, HR is on board with getting to v3. They approve of flipping the theme on DEV or STAGE to get a sense of the change and to see what breaks. They would also be interested in access to a v3 site to play around with layout builder.

They are open-minded to reducing customizations but will need to have some internal discussions related to:

- FAQs -> Collections
- Related Content -> in-page callouts
- Footer Admin Services -> main menu
- Unit Contact information -> remove entirely or move to block outside of layout builder.
- Interested in main menu visibility options in the future to show different than active-children but would not block go-live.

Might get hung up due to content changes if they decided to do away with customizations.

**Comment by pyrello** [comment]
A quick note that it is trivial to add the Related Content and Unit Contact Information blocks to the page LB template as part of this shift, so that they show up on every page where there are results, similar to how they do now. However, since we have disabled the use of the native block labels for setting headlines (titles) on blocks, we would either need to hard-code these block titles via a block template, or rework our headline implementation to use block labels instead of being a separate field.

**Comment by pyrello** [comment]
Also noting that #1713 and #1907 are both blocked until #3707 (or another PR that accounts from removing the `uiowa_bar` and `uiowa_footer` modules from HR's split) has been merged. This is because the modules have to be present to be uninstalled via removal from the split, so there will need to be a follow-up task to actually remove them.

**Comment by pyrello** [comment]
I re-organized the tasks to delineate items that need to (can) be handled as part of the initial conversion and follow-up tasks. 

The second to last item in that list I'm a bit confused about and since there is no attached issue, I think it needs some more explanation:

> Move sitenow-events/single-event library to uids_base and refactor.

Since @richardbporter is out for a couple weeks, I'm guessing that @joewhitsitt is our best potential source of insight into understanding it.

I also don't understand the last item:

> Remove https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/sitenow_people/sitenow_people.module#L269-L287 per https://github.com/uiowa/uiowa/pull/2906

I don't see anything in the comments of the linked (merged) PR that explains it and the code lines in question seem to be no longer relevant since they are tracking the current version of main instead of using the permalink to the commit that was relevant at the time. I attempted to reconstruct what the lines would have been at the time that the PR was merged, but am not clear on what that indicates either: https://github.com/uiowa/uiowa/blob/c38aa0bcfd8b4c6a696f4ef3cd285c728f4b711e/docroot/modules/custom/sitenow_people/sitenow_people.module#L269-L287

**Comment by joewhitsitt** [comment]
I think these hardcoded heading classes are no longer used in uids_base and can be removed once HR is off uiowa_bootstrap. I remember @bspeare asking me about where the classes were coming from.

**Comment by joewhitsitt** [comment]
Regarding events, uiowa_bootstrap has a different templates for events. Forgot what changed, but I think that JS is serving both versions (different variables?) and could be refactored to remove the old stuff once it is no longer in use. Not sure about the idea of moving it.

**Comment by pyrello** [comment]
Thanks!

It seemed like we only needed to split `config_ignore.settings` because of `uiowa_footer` or `uiowa_bar`, so I removed that in #3709

**Comment by pyrello** [comment]
FWIW, the yarn workspaces are not individually defined and will no longer be active when the associated themes are removed. Removing those will have to happen after #3709.

**Comment by pyrello** [comment]
I have updated the `develop` branch with these changes in #3709, so we are just waiting to get feedback from HR about how to proceed.

**Comment by bspeare** [comment]
It looks like a lot of pages have the "Section Settings" Content Width set to `Large` which makes the pages look like https://hr.dev.drupal.uiowa.edu/pay/pay-plans/merit-lead-status-pay-plan or https://hr.dev.drupal.uiowa.edu/pay/pay-plans/professional-and-scientific-pay-structure or https://hr.dev.drupal.uiowa.edu/development/leadership-development/career-and-communication-coaching/online-mbti-assessment.   Do we need an update hook to switch all pages to use `Medium` width sections?

<img width="1136" alt="Screen Shot 2021-06-28 at 4 29 39 PM" src="https://user-images.githubusercontent.com/1036433/123706560-0c37a280-d82e-11eb-8f9f-46b06dc92261.png">

**Comment by joewhitsitt** [comment]
A lot of pages in DEV are missing the sidebar because that is one thing that is custom to their site. We are waiting to hear what they need going forward before trying to fix pages like these.

https://hr.uiowa.edu/pay/pay-plans/merit-lead-status-pay-plan

**Comment by joewhitsitt** [comment]
HR would like to retain unit contact information block would be open to placing this outside of main content.

They will work on manually converting faqs to accordions but we might want to just style it temporarily to move things along.

They will find new homes for related content links, and during the transition they will find a new home in the main menu for multi-level footer links.

Is there a way we can make DEV more stable for this transition or do we need to shorten the timeline to fix in stage?

**Comment by richardbporter** [comment]
> Is there a way we can make DEV more stable for this transition or do we need to shorten the timeline to fix in stage?

Like, in general? Or is something broken specifically with the HR site in dev?

**Comment by joewhitsitt** [comment]
I rebuilt dev the other day and lost the feature branch for HR.

**Comment by richardbporter** [comment]
Gotcha. Any time we rebuild develop from main we have to re-merge any feature branches we want on there, probably before pushing it.

**Comment by joewhitsitt** [comment]
We need some kind of dashboard that says what branches are in DEV. Maybe Slackbot could be configured to respond to a "does anyone mind if I [rebuild|checkout|wipe] DEV"

**Comment by richardbporter** [comment]
![its-happening](https://media.giphy.com/media/5mBE2MiMVFITS/giphy.gif?cid=ecf05e4774jdaeo2g2fh8vyh6qn1j4lhzouszs7fpxcytxzp&rid=giphy.gif&ct=g)

**Comment by richardbporter** [comment]
Removed block labels from this and related issues.

**Comment by pyrello** [comment]
It appears the changes to `sitenow_people` indicated in the checklist above have already happened.

**Comment by pyrello** [comment]
IMO, the tasks related to moving the `uiowa_auth` module into this repo and moving/refactoring the `single-events` library should not happen as part of the follow-up to getting HR onto UIDS. The first one requires introducing a new CI process. The second one is still not entirely clear to me why it is necessary or desirable so I created a separate issue to allow for better definition/grooming.

**Comment by richardbporter** [comment]
> IMO, the tasks related to moving the `uiowa_auth` module into this repo and moving/refactoring the `single-events` library should not happen as part of the follow-up to getting HR onto UIDS. The first one requires introducing a new CI process. The second one is still not entirely clear to me why it is necessary or desirable so I created a separate issue to allow for better definition/grooming.

That makes sense.

**Comment by richardbporter** [comment]
> > IMO, the tasks related to moving the `uiowa_auth` module into this repo and moving/refactoring the `single-events` library should not happen as part of the follow-up to getting HR onto UIDS. The first one requires introducing a new CI process. The second one is still not entirely clear to me why it is necessary or desirable so I created a separate issue to allow for better definition/grooming.
> 
> That makes sense.

I removed those from the this issue checklist.
